### ISSUE TITLE
'Groups' feature idea

### ISSUE THREAD

**Issue by That-Ski-Freak** [mastodon/mastodon#25983]
### Pitch

I think being able to be in a group of other people, almost in a similar way to how you can be in a discord server, would be quite nice.

Someone could create a group and have other people join it. Give the group its own feed similar to the server/instance local feed, and in the same vein put posts from your groups on your home page. Groups could also be made to be topic specific. This idea is obviously somewhat similar to how the federated servers/instances already exist on mastodon, but there are some very important differences. Lots of info explaining my thoughts below.

### Motivation

When I sign up to mastodon I am immediately confronted with a big issue (for me): it's hard/impossible to get it set up to where I see the things I want to see. Not having an algorithm is great, but there aren't enough people I know on the internet or personalities/creators I follow who exist on mastodon in order to fill a feed with just stuff from people I think post nice stuff. This is kind of an inevitability unless mastodon becomes significantly more popular. 

Tags seek to be the solution here, but my issue with tags is that they don't allow me to follow and see the things I am generally interested in: broadly technology, video game, and media related things. Tags are very precise, if you are making a post related to a certain topic you add in the tag and people who are into that specific thing will see it, but they don't seem to work as well when it comes to categorizing posts into more generalized topics. It's possible that this is simply because people don't tend to bother categorizing their posts in such broad ways with a text hashtag.

I think stuff like mastodon works much better when you have an actual social circle of people you commonly like to see things from and talk to. I have a discord server with ~60 people I know, somewhat large group of internet friends. A lot of us are on twitter, but we don't really use it much for actually talking to people because it sucks for that, mastodon in theory would be better. Now, mastodon has servers/instances, and when you join one you have the local feed, which would be a good way to get everyone into one space and begin a social network. However, running an instance as a random user is not really practical, and you can only be in one.

This is why I suggest what I'm calling 'groups'. If someone could just create a group and have other people join it, give the group its own feed similar to the server local feed, and in the same vein put posts from your groups on your home page, I think that would be really nice. This would really help in being able to get on with a group of other people, and invite new people to the platform giving them an easy place to begin: their friend's group or the group for a certain topic.

Having groups related to topics would also go a long way to improving the sorting of content that tags are already doing. Picture this: someone makes a group for minecraft, or something smaller like trackmania, or something bigger like technology. I know for sure it would be a lot easier to get people from trackmania twitter over to mastodon, for example, if there was a trackmania group they could join. Internet personalities and creators could also make their own groups, similar to how youtubers often have discord servers (I believe once you reach a certain amount of members in a discord server it starts to get worse and worse, so especially with larger servers like those that youtube creators have, having discrete posts on a platform like mastodon would probably be significantly better). Anyway a group for a topic could be set up to only take posts from it's member users if the post has a certain tag, this would work better for bigger stuff. Now, when you go to make a post, you will see buttons for each of your groups, and you can click the ones it is related to in order to easily add it's tag to your post. If groups become this kind of bigger thing I am describing in this block, being able to select which groups you want to appear on your home page and which ones you just want to have available to go to the local page of would probably be important.

As a bit of a side tangent at the end here, it's possible this last thing I mentioned would be a good feature for tags in general. Having some buttons available when you post that display a few common broad topics or tags you commonly use could help with posts for one thing not being tagged with the broader thing it is a sub topic of. You could write your post about idk zelda tears of the kingdom or something, and it will give you a small list of tags, which will remind you that you should probably also tag it with video games and not just totk. I'm not sure if this would really make that massive of a difference, but it's just a small idea at the end here.

I believe this groups feature idea would improve mastodon significantly, and also go a long way to attract new users. I really want to like the platform, but right now I don't think I can use it effectively. I personally think this feature is an amazing idea, and I hope you guys agree and can add it :D

**Comment by trwnh** [comment]
Duplicate of #139

**Comment by That-Ski-Freak** [comment]
I don't fully understand what issue #139 is suggesting because I don't know what bangtags are (or why it is a duplicate because of that, sorry), but it seems like a lot of people it back then so I'm a little disappointed it doesn't exist.
However, that seems to be pretty old, and I see now there is #19059 , so is some kind of group feature going to be added at some point in the future?

**Comment by trwnh** [comment]
Yes, it's a work in progress. It won't be using !bangtags, but rather a separate UI for it. #19059 is the work in progress PR for it; it's currently mostly done for local groups, but otherwise being held up by protocol review and waiting for feedback from other implementations so that it federates well.

**Comment by That-Ski-Freak** [comment]
That's great news, thanks! I hope to see it merged in soon.

**Comment by geonom** [comment]
May I suggest the ability to write privately within groups as well?

That would enable our NGO to actually move to Mastodon as their main communications device :)
